.azure/pipelines/Phonebook.preview.yml

```diff
@@ -0,0 +1,193 @@
+          scriptLocation: inlineScript
+          inlineScript: |
+            az network public-ip create \
+                --resource-group MC_phonebook-kubernetes_phonebook-cluster_northeurope\
+                --name $(azure_public_ip_name)\
+                --allocation-method static
+
+      - task: AzureCLI@1
+        name: get_public_ip
+        displayName: 'Azure CLI '
+        inputs:
+          azureSubscription: 'Visual Studio Enterprise – MPN (44678328-75c5-4e48-92c0-b2ab31d5166c)'
+          scriptLocation: inlineScript
+          inlineScript: |
+            ip=`az network public-ip show --resource-group MC_phonebook-kubernetes_phonebook-cluster_northeurope --name $(azure_public_ip_name) --query ipAddress --output tsv`
+            echo "##vso[task.setvariable variable=public_ip]$ip"
+            echo "##vso[task.setvariable variable=public_ip;isOutput=true]$ip"
+
+      - task: Kubernetes@1
+        displayName: 'Create Namespace'
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceEndpoint: 'Phonebook Kubernetes Demo Cluster'
+          namespace: 'kube-public'
+          command: 'apply'
+          useConfigurationFile: true
+          configurationType: 'inline'
+          inline: |
+            apiVersion: v1
+            kind: Namespace
+            metadata:
+              name: $(namespace)
+
+      - task: HelmDeploy@0
+        displayName: 'helm init'
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceConnection: 'Phonebook Kubernetes Demo Cluster'
+          namespace: $(namespace)
+          command: init
+          arguments: '--node-selectors "beta.kubernetes.io/os"="linux"'
+          tillerNamespace: $(namespace)
+
+      - task: HelmDeploy@0
+        displayName: 'helm upgrade'
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceConnection: 'Phonebook Kubernetes Demo Cluster'
+          namespace: $(namespace)
+          command: upgrade
+          chartType: FilePath
+          chartPath: '$(Pipeline.Workspace)/$(helm_artifact_name)/phonebook-0.1.0.tgz'
+          releaseName: phonebook-$(namespace)
+          overrideValues: 'frontend.image.tag=$(image_tag),frontend.image.repository=danielhabenicht,frontend.image.name=phonebook,traefik.kubernetes.namespaces={$(namespace)},traefik.loadBalancerIP=$(public_ip),host=$(public_ip)'
+          valueFile: 'Phonebook.Demo/values.yml'
+          recreate: true
+          tillerNamespace: $(namespace)
+
+      - task: Kubernetes@1
+        displayName: 'kubectl create assets'
+        inputs:
+          kubernetesServiceEndpoint: 'Phonebook Kubernetes Demo Cluster'
+          namespace: 'kube-public'
+          command: apply
+          arguments: '-f https://raw.githubusercontent.com/T-Systems-MMS/phonebook/master/Phonebook.Demo/Phonebook.Assets.yml'
```

Comment:
is master the right thing for PR preview? Maybe someone make changes to the yml and want test these changes...

Reply:
But think also about security. I don't understand the yml stuff so be careful if it can change request or create workload on external systems...

Reply:
It's previewing every pr with the master branch as target. 
The build is only activated for prs from contributers or via azure pipeline GitHub commands.

Reply:
Uhh, yep forgot to change the paths